storage/storage.go

```diff
@@ -35,3 +52,14 @@ func (f Fanout) Append(s *model.Sample) {
 		a.Append(s)
 	}
 }
+
+// Throttled returns true if at least one of the SampleAppenders in the Fanout
+// slice is throttled.
+func (f Fanout) Throttled() bool {
```

Comment:
> We can throttle if a single one needs it – but that doesn't mean we have to throttle all the other ones.

This is not the way throttling works (at least with the basic idea implemented here). Throttling means that you don't even start a scrape if your storage is not ready to take the result.

> It would be a bunch easier if `Append()` accepted a vector of samples.

Which leads us back to the old discussion why we _removed_ a vector of samples from the Append signature.
* It requires to copy all samples (up to 500k/s) into a vector before handing it over to the storage, with no way of reusing it.
* It suggests atomicity that the current storage is not designed to deliver. (How wold you return the information that for three out of the 1000 samples in the vector, there was an out-of-order timestamp?)
* It does not help with the idea of _first_ finding out if the storage is ready to ingest and _then_ do the work of a potentially very large scrape.